src/matchers/__tests__/to-be-visible.test.tsx

```diff
@@ -0,0 +1,206 @@
+    <View testID="test" aria-hidden={true} />
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+
+  update(<View testID="test" />);
+  expect(getByTestId('test')).toBeVisible();
+});
+
+test('toBeVisible() on view within inaccessible view', () => {
+  const { getByTestId } = render(
+    <View aria-hidden={true}>
+      <View>
+        <View testID="test" />
+      </View>
+    </View>
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+});
+
+test('toBeVisible() on inaccessible view (iOS)', () => {
+  const { getByTestId, update } = render(
+    <View testID="test" accessibilityElementsHidden />
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+
+  update(<View testID="test" accessibilityElementsHidden={false} />);
+  expect(getByTestId('test')).toBeVisible();
+});
+
+test('toBeVisible() on view within inaccessible view (iOS)', () => {
+  const { getByTestId } = render(
+    <View accessibilityElementsHidden>
+      <View>
+        <View testID="test" />
+      </View>
+    </View>
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+});
+
+test('toBeVisible() on inaccessible view (Android)', () => {
+  const { getByTestId, update } = render(
+    <View testID="test" importantForAccessibility="no-hide-descendants" />
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+
+  update(<View testID="test" importantForAccessibility="auto" />);
+  expect(getByTestId('test')).toBeVisible();
+});
+
+test('toBeVisible() on view within inaccessible view (Android)', () => {
+  const { getByTestId } = render(
+    <View importantForAccessibility="no-hide-descendants">
+      <View>
+        <View testID="test" />
+      </View>
+    </View>
+  );
+  expect(
+    getByTestId('test', { includeHiddenElements: true })
+  ).not.toBeVisible();
+});
+
+test('toBeVisible() on null elements', () => {
+  expect(() => expect(null).toBeVisible()).toThrowErrorMatchingInlineSnapshot(`
+      "expect(received).toBeVisible()
+
+      received value must be a host element.
+      Received has value: null"
+    `);
+});
+
+test('toBeVisible() on non-React elements', () => {
+  expect(() =>
+    expect({ name: 'Non-React element' }).not.toBeVisible()
+  ).toThrow();
+  expect(() => expect(true).not.toBeVisible()).toThrow();
+});
+
+test('toBeVisible() throws an error when expectation is not matched', () => {
+  const { getByTestId, update } = render(<View testID="test" />);
+  expect(() =>
+    expect(getByTestId('test')).not.toBeVisible()
+  ).toThrowErrorMatchingSnapshot();
+
+  update(<View testID="test" style={{ opacity: 0 }} />);
+  expect(() =>
+    expect(getByTestId('test')).toBeVisible()
+  ).toThrowErrorMatchingSnapshot();
+});
+
+test('toBeVisible() on Pressable with function style prop', () => {
```

Comment:
What is the purpose of this test?

Reply:
To test that the matcher still works when passing `style` as a function, which is exclusive to the `Pressable` component. But I can also see the point for removing it :)

Reply:
AFAIK that style as a function is working on Pressable composite element level, but get's transformed to style object on the underlying host View element.

Reply:
Yep! Ok, I'll remove it